types/yandex-maps/index.d.ts

```diff
@@ -0,0 +1,3043 @@
+				addViaPoints?: boolean;
+				addWayPoints?: boolean;
+				editViaPoints?: boolean;
+				editWayPoints?: boolean;
+				removeViaPoints?: boolean;
+				removeWayPoints?: boolean;
+			}): void;
+
+			stop(): void;
+		}
+
+		export abstract class Route implements IGeoObject {
+			geometry: IGeometry | null;
+			properties: IDataManager;
+			state: IDataManager;
+			events: IEventManager;
+			options: IOptionManager;
+			editor: Editor;
+
+			getOverlay(): Promise<IOverlay | null>;
+
+			getOverlaySync(): IOverlay | null;
+
+			getParent(): null | IControlParent;
+
+			setParent(parent: IControlParent): this;
+
+			getMap(): Map;
+
+			getHumanJamsTime(): string;
+
+			getHumanLength(): string;
+
+			getHumanTime(): string;
+
+			getJamsTime(): number;
+
+			getLength(): number;
+
+			getPaths(): GeoObjectCollection;
+
+			getTime(): number;
+
+			getViaPoints(): GeoObjectCollection;
+
+			getWayPoints(): GeoObjectCollection;
+		}
+	}
+
+	export class Balloon extends Popup<Balloon> implements IBaloon<Balloon> {
+		constructor(map: Map, options?: IBalloonOptions);
+
+		getData(): object;
+
+		close(force?: boolean): Promise<Balloon>;
+
+		getParent(): Balloon | null;
+
+		setParent(parent: Balloon): this;
+
+		autoPan(): Promise<Balloon>;
+
+		freeze(): IFreezable;
+
+		isFrozen(): boolean;
+
+		unfreeze(): IFreezable;
+
+		add(types: string[][] | string[] | string, callback: (event: (object | IEvent)) => void, context?: object, priority?: number): this;
+
+		group(): IEventGroup;
+
+		remove(types: string[][] | string[] | string, callback: (event: (object | IEvent)) => void, context?: object, priority?: number): this;
+
+		fire(type: string, eventobject: object | IEvent): this;
+	}
+
+	interface IBalloonOptions {
+		autoPan?: boolean;
+		autoPanCheckZoomRange?: boolean;
+		autoPanDuration?: number;
+		autoPanMargin?: number[][] | number[] | number;
+		autoPanUseMapMargin?: boolean;
+		closeButton?: boolean;
+		contentLayout?: IClassConstructor<ILayout> | string;
+		layout?: IClassConstructor<ILayout> | string;
+		maxHeight?: number;
+		maxWidth?: number;
+		minHeight?: number;
+		minWidth?: number;
+		offset?: number[];
+		pane?: string;
+		panelContentLayout?: IClassConstructor<ILayout> | string;
+		panelMaxHeightRatio?: number;
+		panelMaxMapArea?: number;
+		shadow?: boolean;
+		shadowLayout?: IClassConstructor<ILayout> | string;
+		shadowOffset?: number[];
+	}
+
+	export class Circle implements GeoObject {
+		constructor(geometry: ICircleGeometry[][][][] | number[][] | object, properties?: object | IDataManager, options?: ICircleOptions)
+
+		balloon: geoObject.Balloon;
+		editor: IGeometryEditor;
+		hint: geoObject.Hint;
+		events: event.Manager;
+		options: option.Manager;
+		properties: data.Manager;
+		state: data.Manager;
+
+		geometry: IGeometry | null;
+		indices: ArrayBuffer;
+		vertices: ArrayBuffer;
+
+		getOverlay(): Promise<IOverlay | null>;
+
+		getOverlaySync(): IOverlay | null;
+
+		getParent(): any | IControlParent;
```

Comment:
I saw many other properties like this, so I'm just commenting on this one:

Using `any` in union types is not a good idea. `any | IControlParent` is still the same type as `any`. Consider just using `any`, just `IControlParent`, or maybe `IControlParent | object` depending on what you're trying to do?

Reply:
Sad for this. This is due WebStorm by default implements interface of T|null to T|any. Fixed.